TooltipDecorator/useTooltip.js

```diff
@@ -0,0 +1,266 @@
+/* global MutationObserver ResizeObserver */
+
+import {forProp, forward, handle, not} from '@enact/core/handle';
+import {Job} from '@enact/core/util';
+import {useI18nContext} from '@enact/i18n/I18nDecorator';
+import {FloatingLayerBase} from '@enact/ui/FloatingLayer';
+import ri from '@enact/ui/resolution';
+
+import {Tooltip, defaultArrowAnchor, defaultDirection} from './Tooltip';
+import {adjustDirection, adjustAnchor, calcOverflow, getLabelOffset, getPosition} from './util';
+
+let currentTooltip; // needed to know whether or not we should stop a showing job when unmounting
+
+import useHandlers from '@enact/core/useHandlers';
+import {useCallback, useEffect, useLayoutEffect, useRef, useState} from 'react';
+
+function getDirectionAnchor (tooltipPosition, tooltipType) {
+	const position = tooltipPosition || (defaultDirection(tooltipType) + ' ' + defaultArrowAnchor(tooltipType));
+	const arr = position.split(' ');
+
+	if (arr.length === 2) {
+		return {
+			tooltipDirection: arr[0],
+			arrowAnchor: arr[1]
+		};
+	} else {
+		return {
+			tooltipDirection: position === 'below' ? 'below' : 'above',
+			arrowAnchor: 'right'
+		};
+	}
+}
+
+const removeTooltipProps = ({...props}) => {
```

Comment:
Why is the parameter `{...props}`? Can we just use `props`?

Reply:
This function's role is to delete useless properties from immutable `props`. If the name 'props' makes misunderstanding, I can rename it to 'rest' or something other. Any suggestion?